### PR TITLE
Revamp of the Stash Origin document to match xcache-1.0 RPM.

### PR DESCRIPTION
**Pull request by bbockelm**
Still needs testing and review -- as far as I could get this evening.

### CODE REVIEW

File: docs/data/stashcache/install-origin.md [hunk 1]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
 ---------------
 
 Before starting the installation process, consider the following points:
 
-* __Operating system:__ A RHEL 7 or compatible operating systems is **strongly** recommended.
-    If you have special needs that require you to run your origin server on a RHEL 6-based host,
-    contact us at support@opensciencegrid.org.
-* __User IDs:__ If they do not exist already, the installation will create the Linux user IDs `condor` and `xrootd`
-* __Host certificate:__ The StashCache server uses a host certificate to advertise to a central collector.
+* __Operating system:__ A RHEL 7 or compatible operating systems.
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]:
```suggestion
* __Operating system:__ A RHEL 7 or compatible operating system.
```

File: docs/data/stashcache/install-origin.md [hunk 2]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
 
-* `oss.localroot /stash`: Change the `localroot` to the location where your data is mounted on
-  the origin server; default is `/stash`.
-* `all.export /<YOUR VO>`: A sub-directory within the `localroot` directory that will be exported.
-  If multiple directories must be exported, you may specify `all.export` multiple times.
+* `origindir`: the mounted filesystem path to export.  This document refers to this as `/stash`.
+* `originexport`: the file within the `origindir` to export.  This MUST be in your supported VO's registered namespace.
+* `sitename`: the resource name registered with OSG.
 
 For example, if the HCC VO would like to set up an origin server exporting from the mountpoint `/mnt/bigdata`,
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]: If we say that we're going to refer to the mount point as `/stash` throughout the document, then we should do so -- i.e. let's replace `/mnt/bigdata` with `/stash` everywhere.

Comment (bbockelm) [reply docs/data/stashcache/install-origin.md]: Done!

**Comment by bbockelm** [comment]
@matyasselmeci - ready for re-review.

File: docs/data/stashcache/install-origin.md [hunk 3]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
 * Install [CA certificates](/common/ca.md)
 
-Installing the StashCache origin
---------------------------------
+Installing the origin
+---------------------
 
-The StashCache daemon consists of an XRootD server and an HTCondor-based service for collecting and reporting
-monitoring data about the origin. To simplify installation, OSG provides convenience RPMs that install all required
+The origin service consists of an XRootD daemons and its dependencies for the authentication infrastructure.
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]:
```suggestion
The origin service consists of one or more XRootD daemons and their dependencies for the authentication infrastructure.
```

File: docs/data/stashcache/install-origin.md [hunk 4]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
-* `all.export /<YOUR VO>`: A sub-directory within the `localroot` directory that will be exported.
-  If multiple directories must be exported, you may specify `all.export` multiple times.
+* `origindir`: the mounted filesystem path to export.  This document refers to this as `/mnt/stash`.
+* `originexport`: the file within the `origindir` to export.  This MUST be in your supported VO's registered namespace.
+* `sitename`: the resource name registered with OSG.
 
-For example, if the HCC VO would like to set up an origin server exporting from the mountpoint `/mnt/bigdata`,
-but only export the subdirectories `/mnt/bigdata/hcc/bio/datasets` and `/mnt/bigdata/hcc/hep/generators`,
-they would use the following configuration:
+For example, if the HCC VO would like to set up an origin server exporting from the mountpoint `/mnt/stash`,
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]:
```suggestion
For example, if the HCC VO would like to set up an origin server exporting from the mountpoint `/stash`,
```

File: docs/data/stashcache/install-origin.md [hunk 5]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
+but HCC's registered namespace is `/hcc`, then the following would be set:
 
 ```
-oss.localroot /mnt/bigdata
-all.export /hcc/bio/datasets
-all.export /hcc/hep/generators
+set origindir = /mnt/stash
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]:
```suggestion
set origindir = /stash
```

File: docs/data/stashcache/install-origin.md [hunk 6]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
+set originexport = /hcc
+set sitename = HCC_STASH_ORIGIN
 ```
 
-With this configuration, the data under `/mnt/bigdata/hcc/bio/datasets` would be available under the StashCache path
-`/hcc/bio/datasets`, the data under `/mnt/bigdata/hcc/hep/generators` would be available under the StashCache path
-`/hcc/hep/generators`, and no other data would be available via StashCache.
+With this configuration, the data under `/mnt/stash/hcc/bio/datasets` would be available under the StashCache path
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]:
```suggestion
With this configuration, the data under `/stash/hcc/bio/datasets` would be available under the StashCache path
```

File: docs/data/stashcache/install-origin.md [hunk 7]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
+`/hcc/bio/datasets` and the data under `/mnt/stash/hcc/hep/generators` would be available under the StashCache path
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]:
```suggestion
`/hcc/bio/datasets` and the data under `/stash/hcc/hep/generators` would be available under the StashCache path
```

File: docs/data/stashcache/install-origin.md [hunk 8]
```diff
@@ -100,11 +102,11 @@ The origin service consists of the following systemd units:
 
 | **Software** | **Service name** | **Notes** |
 |--------------|------------------|-----------|
-| XRootD | `xrootd@stashcache-origin-server.service` | The xrootd daemon, which performs the data transfers |
-| XRootD | `cmsd@stashcache-origin-server.service` | The "cluster management service" daemon, which integrates the origin into the data federation.  |
-| Fetch CRL         | `fetch-crl-boot` and `fetch-crl-cron` | See [CA documentation](/common/ca#managing-fetch-crl-services) for more info |
+| XRootD | `xrootd@stash-origin.service` | The xrootd daemon, which performs the data transfers |
+| XRootD | `xrootd@stash-origin-auth.service` | The xrootd daemon, which performs authenticated data transfers |
+| XRootD | `cmsd@stash-origin.service` | The "cluster management service" daemon, which integrates the origin into the data federation.  |
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]: Do we need a `cmsd@stash-origin-auth.service` as well?

File: docs/data/stashcache/install-origin.md [hunk 9]
```diff
@@ -100,11 +102,11 @@ The origin service consists of the following systemd units:
 
 | **Software** | **Service name** | **Notes** |
 |--------------|------------------|-----------|
-| XRootD | `xrootd@stashcache-origin-server.service` | The xrootd daemon, which performs the data transfers |
-| XRootD | `cmsd@stashcache-origin-server.service` | The "cluster management service" daemon, which integrates the origin into the data federation.  |
-| Fetch CRL         | `fetch-crl-boot` and `fetch-crl-cron` | See [CA documentation](/common/ca#managing-fetch-crl-services) for more info |
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]: We don't need fetch-crl anymore?

File: docs/data/stashcache/install-origin.md (hunk 4 above)
Comment (bbockelm) [reply docs/data/stashcache/install-origin.md]: Yeah, that's what I did originally -- however, I'd like to keep `/mnt` to emphasize that this is on the local filesystem.  I don't think most admins would mount their actual filesystem inside `/`.

File: docs/data/stashcache/install-origin.md (hunk 8 above)
Comment (bbockelm) [reply docs/data/stashcache/install-origin.md]: Nope - the `cmsd` should export both.

(We might want to revisit this, however... this implies that the `xrootd@stash-origin` is running)

File: docs/data/stashcache/install-origin.md (hunk 9 above)
Comment (bbockelm) [reply docs/data/stashcache/install-origin.md]: Admins don't need to start it as systemd handles the dependency.

File: docs/data/stashcache/install-origin.md (hunk 4 above)
Comment (matyasselmeci) [reply docs/data/stashcache/install-origin.md]: If you want to keep `/mnt`, then we need to change it everywhere else in the document and the default values for `cachedir` and `origindir` in the `*-site-local.cfg` files in xcache.

Comment (matyasselmeci) [reply docs/data/stashcache/install-origin.md]: Not that I'm not willing to do that -- we haven't tagged xcache 1.0.1 and if you think /mnt/stash is a better value then we can use it, but it's nice to have documents consistent with the config... and uh they should be consistent with themselves, too :)

Comment (bbockelm) [reply docs/data/stashcache/install-origin.md]: Good point.  Let's change the default suggested `cachedir` and `origindir`.

(Is there somewhere else in this document I missed?)

File: docs/data/stashcache/install-origin.md [hunk 10]
```diff
@@ -11,87 +11,89 @@ to export its data to the StashCache data federation.
...
 ```console
-root@host # yum install stashcache-origin-server
+root@host # yum install --enablerepo=osg-development stash-origin
 ```
 
+!!! note
+    This document covers a completely overhauled stash origin service that depends on XRootD 4.9.0; while this
+    has not yet been released into the OSG production repository, we strongly recommend using this version.
+
 For this installation guide, we assume that the data to be exported to the federation is mounted at `/stash`
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]:
```suggestion
For this installation guide, we assume that the data to be exported to the federation is mounted at `/mnt/stash`
```

File: docs/data/stashcache/install-origin.md (hunk 4 above)
Comment (matyasselmeci) [reply docs/data/stashcache/install-origin.md]: Just at the top. I made a suggestion.

File: docs/data/stashcache/install-origin.md (hunk 8 above)
Comment (matyasselmeci) [reply docs/data/stashcache/install-origin.md]: So something like
```
!!! note
    Make sure `xrootd@stash-origin.service` is running before starting `cmsd@stash-origin.service`
```

Comment (matyasselmeci) [reply docs/data/stashcache/install-origin.md]: And tweaking the description for cmsd:  The "cluster management service" daemon, which integrates the xrootd daemon(s) into the data federation

**Comment by matyasselmeci** [comment]
Other than potential tweaks to the services section, this looks good to me. Feel free to merge with or without the tweaks; I've also been working on changes to the install docs so I'd like to rebase my changes on yours.

**Comment by brianhlin** [comment]
Just a reminder that we should hold off on merging these instructions until we release XCache 1.0+

**Comment by matyasselmeci** [comment]
We're going to need additional updates for the changes introduced in xcache-1.0.1 -- the config tweaks, the service tweaks, and the Authfile generation.

**Comment by matyasselmeci** [comment]
Resolved by #527.

**Pull request closed without merging**